### PR TITLE
Phase2-hgx336C Try to introduce the guard rings for both full and partial wafers of HGCal (will be a part of V18)

### PR DESCRIPTION
**Pull request by bsunanda**
#### PR description:

Try to introduce the guard rings for both full and partial wafers of HGCal (will be a part of V18). 

#### PR validation:

Use the runTheMatrix test workflows

#### If this PR is a backport please specify the original PR and why you need to backport that PR. If this PR will be backported please specify to which release cycle the backport is meant for:

Nothing special

### CODE REVIEW

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by srimanob** [comment]
To confirm from the PR description, when you said it will be part of V18. But the PR touches v16, v17 by adding UseOffset which should not make the change in physics, but SIM history. So that why we see the difference in D88 in PR test.

Do I understand correctly?

**Comment by bsunanda** [comment]
@srimanob Yes it changes SIM history only and that is why we see some difference

**Comment by srimanob** [comment]
+Upgrade

File: Geometry/HGCalCommonData/src/HGCalParametersFromDD.cc [hunk 1]
```diff
@@ -8,7 +8,7 @@
 #include "Geometry/HGCalCommonData/interface/HGCalGeometryMode.h"
 #include "Geometry/HGCalCommonData/interface/HGCalParameters.h"
 
-//#define EDM_ML_DEBUG
+#define EDM_ML_DEBUG
```
Comment (perrotta) [inline Geometry/HGCalCommonData/src/HGCalParametersFromDD.cc:11]: Should this get re-commented out?

**Comment by perrotta** [comment]
+1

**Review by perrotta — COMMENTED** [review]
I missed this one: I think this should be re-commented out, if we don't want to flood the outputs with these log messages. Could you please confirm, Sunanda?

**Comment by perrotta** [comment]
Ping @bsunanda

**Pull request merged**